tests/Proto.Persistence.Tests/InMemoryProvider.cs

```diff
@@ -30,15 +30,17 @@ public Task<(object Snapshot, long Index)> GetSnapshotAsync(string actorName)
 
         public Task<long> GetEventsAsync(string actorName, long indexStart, long indexEnd, Action<object> callback)
         {
+            var lastIndex = 0l;
             if (_events.TryGetValue(actorName, out var events))
             {
                 foreach (var e in events.Where(e => e.Key >= indexStart && e.Key <= indexEnd))
                 {
+                    lastIndex = e.Key;
                     callback(e.Value);
                 }
             }
 
-            return Task.FromResult(0L);
+            return Task.FromResult(lastIndex);
```

Comment:
I don't think this causes any side effects since the `InMemoryProvider` was only used in the unit tests. But, this helped me to write a better test for my domain model object.